src/Controller/OgAdminMembersController.php

```diff
@@ -21,13 +54,82 @@ class OgAdminMembersController extends ControllerBase {
    *   The members overview View.
    */
   public function membersList(RouteMatchInterface $route_match) {
-    $parameter_name = $route_match->getRouteObject()->getOption('_og_entity_type_id');
-
-    /** @var \Drupal\Core\Entity\EntityInterface $group */
-    $group = $route_match->getParameter($parameter_name);
-
+    $group_type_id = $route_match->getRouteObject()->getOption('_og_entity_type_id');
+    $group = $route_match->getParameter($group_type_id);
     $arguments = [$group->getEntityTypeId(), $group->id()];
     return Views::getView('og_members_overview')->executeDisplay('default', $arguments);
   }
 
+  /**
+   * Displays add membership links for available membership types.
+   *
+   * Returns default membership type if that's all that exists.
+   *
+   * @return array
+   *   A render array for a list of the og membership types that can be added;
+   *   however, if there is only one og membership type defined for the site,
+   *   the function will return the default add member form.
+   */
+  public function addPage(RouteMatchInterface $route_match) {
+    $entity_type_id = $route_match->getRouteObject()
+      ->getOption('_og_entity_type_id');
+
+    $group = $route_match->getParameter($entity_type_id);
+
+    $membership_types = $this->entityTypeManager
+      ->getStorage('og_membership_type')
+      ->loadMultiple();
+
+    if ($membership_types && count($membership_types) == 1) {
+      return $this->addForm($group, $membership_types[OgMembershipInterface::TYPE_DEFAULT]);
+    }
+
+    $build = [
+      '#theme' => 'entity_add_list',
```

Comment:
This theme template expects to find a description for each bundle entity. 

Currently, it produces this error:

> Undefined index: description in template_preprocess_entity_add_list()

Reply:
@joachim-n, or add '' instead of NULL?

Reply:
This has been resolved:
```
      $build['#bundles'][$membership_type_id] = [
        'label' => $og_membership_type->label(),
        'description' => NULL,
        'add_link' => Link::createFromRoute($og_membership_type->label(), 'entity.og_membership.add_form', $add_link_params),
      ];

```